NPE in fixing of LineLength Rule

## Describe the bug
If LineLength Rule is fixing annotation length, then NPE happens

## Invalid line
```kotlin
 @Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' and test_execution.test_suite_execution_id = ?3 limit ?1 offset ?2", nativeQuery = true)
 fun retrieveBatches(limit: Int, offset: Int, executionId: Long): List<Test>
```
## Stack trace
```
[ERROR] 2021-04-23 15:57:41 Internal error has occurred in line-length. Please make an issue on this bug at https://github.com/cqfn/diKTat/.
java.lang.NullPointerException
	at org.cqfn.diktat.ruleset.rules.chapter3.LineLength.fixError(LineLength.kt:192)
	at org.cqfn.diktat.ruleset.rules.chapter3.LineLength.access$fixError(LineLength.kt:59)
	at org.cqfn.diktat.ruleset.rules.chapter3.LineLength$checkLength$1$1.invoke(LineLength.kt:92)
	at org.cqfn.diktat.ruleset.rules.chapter3.LineLength$checkLength$1$1.invoke(LineLength.kt:91)
	at org.cqfn.diktat.ruleset.constants.Warnings.warnAndFix(Warnings.kt:255)
	at org.cqfn.diktat.ruleset.rules.chapter3.LineLength.checkLength(LineLength.kt:89)
	at org.cqfn.diktat.ruleset.rules.chapter3.LineLength.logic(LineLength.kt:73)
	at org.cqfn.diktat.ruleset.rules.DiktatRule.visit(DiktatRule.kt:45)
	at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:271)
	at com.pinterest.ktlint.core.KtLint$format$1.invoke(KtLint.kt:23)
	at com.pinterest.ktlint.core.KtLint$visitor$2$2.invoke(KtLint.kt:212)
	at com.pinterest.ktlint.core.KtLint$visitor$2$2.invoke(KtLint.kt:23)
	at com.pinterest.ktlint.core.ast.PackageKt.visit(package.kt:224)
	at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:211)
	at com.pinterest.ktlint.core.KtLint$visitor$2.invoke(KtLint.kt:23)
	at com.pinterest.ktlint.core.KtLint.format(KtLint.kt:264)
	at com.pinterest.ktlint.internal.FileUtilsKt.formatFile(FileUtils.kt:79)
	at com.pinterest.ktlint.KtlintCommandLine.process(Main.kt:331)
	at com.pinterest.ktlint.KtlintCommandLine.access$process(Main.kt:116)
	at com.pinterest.ktlint.KtlintCommandLine$lintFiles$2$1.call(Main.kt:257)
	at com.pinterest.ktlint.KtlintCommandLine$lintFiles$2$1.call(Main.kt:116)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

## Environment information
* build tool (maven/gradle): gradle
* how is diktat run (CLI, plugin, etc.): plugin
* kotlin version: 1.4.32

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -1,5 +1,6 @@
 package org.cqfn.diktat.ruleset.rules.chapter3
 
+import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
 import org.cqfn.diktat.common.config.rules.RuleConfiguration
 import org.cqfn.diktat.common.config.rules.RulesConfig
 import org.cqfn.diktat.common.config.rules.getRuleConfig
@@ -28,6 +29,7 @@ import com.pinterest.ktlint.core.ast.ElementType.KDOC_TEXT
 import com.pinterest.ktlint.core.ast.ElementType.LITERAL_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.LONG_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.LPAR
+import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
 import com.pinterest.ktlint.core.ast.ElementType.OPERATION_REFERENCE
 import com.pinterest.ktlint.core.ast.ElementType.PACKAGE_DIRECTIVE
 import com.pinterest.ktlint.core.ast.ElementType.PARENTHESIZED
@@ -89,7 +91,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
                     LONG_LINE.warnAndFix(configRules, emitWarn, isFixMode,
                         "max line length ${configuration.lineLength}, but was ${line.length}",
                         offset + node.startOffset, node, fixableType != LongLineFixableCases.None) {
-                        fixError(fixableType)
+                        fixError(fixableType, configuration)
                     }
                 }
             }
@@ -187,9 +189,9 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     @Suppress("UnsafeCallOnNullableType", "WHEN_WITHOUT_ELSE")
-    private fun fixError(fixableType: LongLineFixableCases) {
+    private fun fixError(fixableType: LongLineFixableCases, configuration: LineLengthConfiguration) {
         when (fixableType) {
-            is LongLineFixableCases.Fun -> fixableType.node.appendNewlineMergingWhiteSpace(null, fixableType.node.findChildByType(EQ)!!.treeNext)
+            is LongLineFixableCases.Fun -> fixFunction(fixableType.node, configuration)
             is LongLineFixableCases.Comment -> fixComment(fixableType)
             is LongLineFixableCases.Condition -> fixLongBinaryExpression(fixableType)
             is LongLineFixableCases.Property -> createSplitProperty(fixableType)
@@ -197,6 +199,49 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             is LongLineFixableCases.None -> return
         }
     }
+
+    /**
+     * This function handles two cases:
+     *      1. When fun foo() = ...
+     *      2. There is an annotation before the function which breaks the maxLength parameter.
+     */
+    private fun fixFunction(wrongFunction: ASTNode, configuration: LineLengthConfiguration) {
+        if (wrongFunction.findChildByType(EQ) != null) {
+            wrongFunction.appendNewlineMergingWhiteSpace(null, wrongFunction.findChildByType(EQ)!!.treeNext)
+            return
+        } else if (wrongFunction.findChildByType(MODIFIER_LIST) != null) {
+            val annotationText = fixAnnotation(wrongFunction.findChildByType(MODIFIER_LIST)!!, configuration)
+            // The last one will always be fun declaration as there is no block.
+            val funText = wrongFunction.text.lines().last()
+            wrongFunction.treeParent.replaceChild(wrongFunction, KotlinParser().createNode("$annotationText\n$funText"))
+        }
+    }
+
+    private fun fixAnnotation(node: ASTNode, configuration: LineLengthConfiguration): String {
+        val annotationEntry = node.findChildByType(ANNOTATION_ENTRY)!!
+        val annotationText = annotationEntry.text
+        val firstPart = annotationText.substring(0, configuration.lineLength.toInt())
+        val remainingPart = annotationText.substring(configuration.lineLength.toInt(), annotationText.length)
+        return if (isDividerInString(firstPart)) {
+            divideString(firstPart) + remainingPart
+        } else {
+            divideAnnotationByComma(firstPart) + remainingPart
+        }
+    }
+
+    private fun divideAnnotationByComma(text: String): String {
+        val lastIndex = text.indexOfLast { it == ',' } + 1 // plus one is to include comma on the first line
+        return text.substring(0, lastIndex)  + "\n" + text.substring(lastIndex, text.length)
+    }
+
+    private fun divideString(text: String): String {
+        val divisionIndex = text.indexOfLast { it == ' ' }
+        return text.substring(0, divisionIndex) + "\" +\n\"" + text.substring(divisionIndex, text.length)
+    }
+
+    private fun isDividerInString(text: String): Boolean =
+        text.count { it == '"' } % 2 != 0
+
 
     private fun fixComment(wrongComment: LongLineFixableCases.Comment) {
         val wrongNode = wrongComment.node

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -1,6 +1,5 @@
 package org.cqfn.diktat.ruleset.rules.chapter3
 
-import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
 import org.cqfn.diktat.common.config.rules.RuleConfiguration
 import org.cqfn.diktat.common.config.rules.RulesConfig
 import org.cqfn.diktat.common.config.rules.getRuleConfig
@@ -11,6 +10,7 @@ import org.cqfn.diktat.ruleset.utils.appendNewlineMergingWhiteSpace
 import org.cqfn.diktat.ruleset.utils.calculateLineColByOffset
 import org.cqfn.diktat.ruleset.utils.hasChildOfType
 
+import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.BINARY_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.BLOCK
 import com.pinterest.ktlint.core.ast.ElementType.BOOLEAN_CONSTANT
@@ -230,7 +230,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     private fun divideAnnotationByComma(text: String): String {
-        val lastIndex = text.indexOfLast { it == ',' } + 1 //  plus one is to include comma on the first line
+        val lastIndex = text.indexOfLast { it == ',' } + 1  // plus one is to include comma on the first line
         return text.substring(0, lastIndex) + "\n" + text.substring(lastIndex, text.length)
     }
 
@@ -239,9 +239,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         return text.substring(0, divisionIndex) + "\" +\n\"" + text.substring(divisionIndex, text.length)
     }
 
-    private fun isDividerInString(text: String): Boolean =
-            text.count { it == '"' } % 2 != 0
-
+    private fun isDividerInString(text: String): Boolean = text.count { it == '"' } % 2 != 0
 
     private fun fixComment(wrongComment: LongLineFixableCases.Comment) {
         val wrongNode = wrongComment.node

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -230,8 +230,8 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     private fun divideAnnotationByComma(text: String): String {
-        val lastIndex = text.indexOfLast { it == ',' } + 1 // plus one is to include comma on the first line
-        return text.substring(0, lastIndex)  + "\n" + text.substring(lastIndex, text.length)
+        val lastIndex = text.indexOfLast { it == ',' } + 1 //  plus one is to include comma on the first line
+        return text.substring(0, lastIndex) + "\n" + text.substring(lastIndex, text.length)
     }
 
     private fun divideString(text: String): String {
@@ -240,7 +240,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     private fun isDividerInString(text: String): Boolean =
-        text.count { it == '"' } % 2 != 0
+            text.count { it == '"' } % 2 != 0
 
 
     private fun fixComment(wrongComment: LongLineFixableCases.Comment) {

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -1,6 +1,5 @@
 package org.cqfn.diktat.ruleset.rules.chapter3
 
-import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
 import org.cqfn.diktat.common.config.rules.RuleConfiguration
 import org.cqfn.diktat.common.config.rules.RulesConfig
 import org.cqfn.diktat.common.config.rules.getRuleConfig
@@ -9,8 +8,10 @@ import org.cqfn.diktat.ruleset.rules.DiktatRule
 import org.cqfn.diktat.ruleset.utils.KotlinParser
 import org.cqfn.diktat.ruleset.utils.appendNewlineMergingWhiteSpace
 import org.cqfn.diktat.ruleset.utils.calculateLineColByOffset
+import org.cqfn.diktat.ruleset.utils.findParentNodeWithSpecificType
 import org.cqfn.diktat.ruleset.utils.hasChildOfType
 
+import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.BINARY_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.BOOLEAN_CONSTANT
 import com.pinterest.ktlint.core.ast.ElementType.CALL_EXPRESSION
@@ -43,7 +44,6 @@ import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
 import com.pinterest.ktlint.core.ast.nextSibling
 import com.pinterest.ktlint.core.ast.parent
 import com.pinterest.ktlint.core.ast.prevSibling
-import org.cqfn.diktat.ruleset.utils.findParentNodeWithSpecificType
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
 import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement
 import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -15,7 +15,6 @@ import com.pinterest.ktlint.core.ast.ElementType.BLOCK
 import com.pinterest.ktlint.core.ast.ElementType.BOOLEAN_CONSTANT
 import com.pinterest.ktlint.core.ast.ElementType.CALL_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.CHARACTER_CONSTANT
-import com.pinterest.ktlint.core.ast.ElementType.CLOSING_QUOTE
 import com.pinterest.ktlint.core.ast.ElementType.CONDITION
 import com.pinterest.ktlint.core.ast.ElementType.EOL_COMMENT
 import com.pinterest.ktlint.core.ast.ElementType.EQ
@@ -31,7 +30,6 @@ import com.pinterest.ktlint.core.ast.ElementType.LITERAL_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.LONG_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.LPAR
 import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
-import com.pinterest.ktlint.core.ast.ElementType.OPEN_QUOTE
 import com.pinterest.ktlint.core.ast.ElementType.OPERATION_REFERENCE
 import com.pinterest.ktlint.core.ast.ElementType.PACKAGE_DIRECTIVE
 import com.pinterest.ktlint.core.ast.ElementType.PARENTHESIZED
@@ -39,7 +37,6 @@ import com.pinterest.ktlint.core.ast.ElementType.POSTFIX_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.PREFIX_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.PROPERTY
 import com.pinterest.ktlint.core.ast.ElementType.REFERENCE_EXPRESSION
-import com.pinterest.ktlint.core.ast.ElementType.REGULAR_STRING_PART
 import com.pinterest.ktlint.core.ast.ElementType.RPAR
 import com.pinterest.ktlint.core.ast.ElementType.SHORT_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.STRING_TEMPLATE
@@ -48,8 +45,6 @@ import com.pinterest.ktlint.core.ast.nextSibling
 import com.pinterest.ktlint.core.ast.parent
 import com.pinterest.ktlint.core.ast.prevSibling
 import org.cqfn.diktat.ruleset.utils.findParentNodeWithSpecificType
-import org.cqfn.diktat.ruleset.utils.hasParent
-import org.jetbrains.kotlin.codegen.writeKotlinMetadata
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
 import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement
 import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
@@ -147,14 +142,17 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         if (difference > node.text.length) {
             return LongLineFixableCases.BinaryExpression(node.treeParent)
         }
-        var delimiterIndex = node.text.substring(0, configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
+        val delimiterIndex = node.text.substring(0, configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
         if (delimiterIndex == -1) {
             return LongLineFixableCases.None
         }
-        if (leftOffset + delimiterIndex > configuration.lineLength.toInt() - 2) {
-            delimiterIndex = node.text.substring(0, delimiterIndex - 2).lastIndexOf(' ')
+        // minus 2 here as we are inserting ` +` and we don't want it to exceed line length
+        val correcterDelimiter = if (leftOffset + delimiterIndex > configuration.lineLength.toInt() - 2) {
+            node.text.substring(0, delimiterIndex - 2).lastIndexOf(' ')
+        } else {
+            delimiterIndex
         }
-        return LongLineFixableCases.StringTemplate(node, delimiterIndex)
+        return LongLineFixableCases.StringTemplate(node, correcterDelimiter)
     }
 
     private fun checkFun(wrongNode: ASTNode) =
@@ -249,6 +247,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         }
     }
 
+    @Suppress("UnsafeCallOnNullableType")
     private fun fixBinaryExpression(node: ASTNode) {
         val whiteSpaceAfterPlus = node.findChildByType(OPERATION_REFERENCE)!!.treeNext
         node.replaceChild(whiteSpaceAfterPlus, PsiWhiteSpaceImpl("\n"))

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -1,5 +1,6 @@
 package org.cqfn.diktat.ruleset.rules.chapter3
 
+import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
 import org.cqfn.diktat.common.config.rules.RuleConfiguration
 import org.cqfn.diktat.common.config.rules.RulesConfig
 import org.cqfn.diktat.common.config.rules.getRuleConfig
@@ -11,7 +12,6 @@ import org.cqfn.diktat.ruleset.utils.calculateLineColByOffset
 import org.cqfn.diktat.ruleset.utils.hasChildOfType
 
 import com.pinterest.ktlint.core.ast.ElementType.BINARY_EXPRESSION
-import com.pinterest.ktlint.core.ast.ElementType.BLOCK
 import com.pinterest.ktlint.core.ast.ElementType.BOOLEAN_CONSTANT
 import com.pinterest.ktlint.core.ast.ElementType.CALL_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.CHARACTER_CONSTANT
@@ -29,7 +29,6 @@ import com.pinterest.ktlint.core.ast.ElementType.KDOC_TEXT
 import com.pinterest.ktlint.core.ast.ElementType.LITERAL_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.LONG_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.LPAR
-import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
 import com.pinterest.ktlint.core.ast.ElementType.OPERATION_REFERENCE
 import com.pinterest.ktlint.core.ast.ElementType.PACKAGE_DIRECTIVE
 import com.pinterest.ktlint.core.ast.ElementType.PARENTHESIZED
@@ -114,9 +113,10 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
                     // 2. FUN with EQ, it seems that new line should be inserted after `=`
                     parent.findParentNodeWithSpecificType(IF)?.let {
                         parent = parent.treeParent
-                    } ?: parent.findParentNodeWithSpecificType(FUN)?.let {
-                        if (it.hasChildOfType(EQ)) {
-                            parent = it
+                    } ?: parent.findParentNodeWithSpecificType(FUN)?.let { node ->
+                        // checking that string template is not in annotation
+                        if (node.hasChildOfType(EQ) && !wrongNode.parents().any { it.elementType == ANNOTATION_ENTRY }) {
+                            parent = node
                         } else {
                             return checkStringTemplate(parent, configuration)
                         }
@@ -132,7 +132,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
      * This class finds where the string can be split
      *
      * @return StringTemplate - if the string can be split,
-     *         BinaryExpression - if there is two concatenated strings and new line should be inserted before `+`
+     *         BinaryExpression - if there is two concatenated strings and new line should be inserted after `+`
      *         None - if the string can't be split
      */
     private fun checkStringTemplate(node: ASTNode, configuration: LineLengthConfiguration): LongLineFixableCases {
@@ -156,7 +156,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     private fun checkFun(wrongNode: ASTNode) =
-            if (!wrongNode.hasChildOfType(BLOCK) && !wrongNode.hasChildOfType(MODIFIER_LIST)) LongLineFixableCases.Fun(wrongNode) else LongLineFixableCases.None
+            if (wrongNode.hasChildOfType(EQ)) LongLineFixableCases.Fun(wrongNode) else LongLineFixableCases.None
 
     private fun checkComment(wrongNode: ASTNode, configuration: LineLengthConfiguration): LongLineFixableCases {
         val leftOffset = positionByOffset(wrongNode.startOffset).second

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -439,12 +439,6 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             val node: ASTNode,
             val indexLastSpace: Int,
             val text: String) : LongLineFixableCases()
-
-        class PropertyWithTemplateEntry(
-            val node: ASTNode,
-            val maximumLineLength: Long,
-            val leftOffset: Int,
-            val binList: MutableList<ASTNode>) : LongLineFixableCases()
     }
 
     /**

--- a/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
+++ b/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt
@@ -10,18 +10,19 @@ import org.cqfn.diktat.ruleset.utils.appendNewlineMergingWhiteSpace
 import org.cqfn.diktat.ruleset.utils.calculateLineColByOffset
 import org.cqfn.diktat.ruleset.utils.hasChildOfType
 
-import com.pinterest.ktlint.core.ast.ElementType.ANNOTATION_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.BINARY_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.BLOCK
 import com.pinterest.ktlint.core.ast.ElementType.BOOLEAN_CONSTANT
 import com.pinterest.ktlint.core.ast.ElementType.CALL_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.CHARACTER_CONSTANT
+import com.pinterest.ktlint.core.ast.ElementType.CLOSING_QUOTE
 import com.pinterest.ktlint.core.ast.ElementType.CONDITION
 import com.pinterest.ktlint.core.ast.ElementType.EOL_COMMENT
 import com.pinterest.ktlint.core.ast.ElementType.EQ
 import com.pinterest.ktlint.core.ast.ElementType.FILE
 import com.pinterest.ktlint.core.ast.ElementType.FLOAT_CONSTANT
 import com.pinterest.ktlint.core.ast.ElementType.FUN
+import com.pinterest.ktlint.core.ast.ElementType.IF
 import com.pinterest.ktlint.core.ast.ElementType.IMPORT_LIST
 import com.pinterest.ktlint.core.ast.ElementType.INTEGER_CONSTANT
 import com.pinterest.ktlint.core.ast.ElementType.KDOC_MARKDOWN_INLINE_LINK
@@ -30,6 +31,7 @@ import com.pinterest.ktlint.core.ast.ElementType.LITERAL_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.LONG_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.LPAR
 import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
+import com.pinterest.ktlint.core.ast.ElementType.OPEN_QUOTE
 import com.pinterest.ktlint.core.ast.ElementType.OPERATION_REFERENCE
 import com.pinterest.ktlint.core.ast.ElementType.PACKAGE_DIRECTIVE
 import com.pinterest.ktlint.core.ast.ElementType.PARENTHESIZED
@@ -37,6 +39,7 @@ import com.pinterest.ktlint.core.ast.ElementType.POSTFIX_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.PREFIX_EXPRESSION
 import com.pinterest.ktlint.core.ast.ElementType.PROPERTY
 import com.pinterest.ktlint.core.ast.ElementType.REFERENCE_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.REGULAR_STRING_PART
 import com.pinterest.ktlint.core.ast.ElementType.RPAR
 import com.pinterest.ktlint.core.ast.ElementType.SHORT_STRING_TEMPLATE_ENTRY
 import com.pinterest.ktlint.core.ast.ElementType.STRING_TEMPLATE
@@ -44,6 +47,9 @@ import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
 import com.pinterest.ktlint.core.ast.nextSibling
 import com.pinterest.ktlint.core.ast.parent
 import com.pinterest.ktlint.core.ast.prevSibling
+import org.cqfn.diktat.ruleset.utils.findParentNodeWithSpecificType
+import org.cqfn.diktat.ruleset.utils.hasParent
+import org.jetbrains.kotlin.codegen.writeKotlinMetadata
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
 import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement
 import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
@@ -91,7 +97,7 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
                     LONG_LINE.warnAndFix(configRules, emitWarn, isFixMode,
                         "max line length ${configuration.lineLength}, but was ${line.length}",
                         offset + node.startOffset, node, fixableType != LongLineFixableCases.None) {
-                        fixError(fixableType, configuration)
+                        fixError(fixableType)
                     }
                 }
             }
@@ -107,14 +113,52 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
                 CONDITION -> return checkCondition(parent, configuration)
                 PROPERTY -> return checkProperty(parent, configuration)
                 EOL_COMMENT -> return checkComment(parent, configuration)
+                STRING_TEMPLATE -> {
+                    // as we are going from bottom to top we are excluding
+                    // 1. IF, because it seems that string template is in condition
+                    // 2. FUN with EQ, it seems that new line should be inserted after `=`
+                    parent.findParentNodeWithSpecificType(IF)?.let {
+                        parent = parent.treeParent
+                    } ?: parent.findParentNodeWithSpecificType(FUN)?.let {
+                        if (it.hasChildOfType(EQ)) {
+                            parent = it
+                        } else {
+                            return checkStringTemplate(parent, configuration)
+                        }
+                    } ?: return checkStringTemplate(parent, configuration)
+                }
                 else -> parent = parent.treeParent
             }
         } while (parent.treeParent != null)
         return LongLineFixableCases.None
     }
 
+    /**
+     * This class finds where the string can be split
+     *
+     * @return StringTemplate - if the string can be split,
+     *         BinaryExpression - if there is two concatenated strings and new line should be inserted before `+`
+     *         None - if the string can't be split
+     */
+    private fun checkStringTemplate(node: ASTNode, configuration: LineLengthConfiguration): LongLineFixableCases {
+        val leftOffset = positionByOffset(node.startOffset).second
+        val difference = configuration.lineLength.toInt() - leftOffset
+        // case when new line should be inserted after `+`. Example: "first" + "second"
+        if (difference > node.text.length) {
+            return LongLineFixableCases.BinaryExpression(node.treeParent)
+        }
+        var delimiterIndex = node.text.substring(0, configuration.lineLength.toInt() - leftOffset).lastIndexOf(' ')
+        if (delimiterIndex == -1) {
+            return LongLineFixableCases.None
+        }
+        if (leftOffset + delimiterIndex > configuration.lineLength.toInt() - 2) {
+            delimiterIndex = node.text.substring(0, delimiterIndex - 2).lastIndexOf(' ')
+        }
+        return LongLineFixableCases.StringTemplate(node, delimiterIndex)
+    }
+
     private fun checkFun(wrongNode: ASTNode) =
-            if (!wrongNode.hasChildOfType(BLOCK)) LongLineFixableCases.Fun(wrongNode) else LongLineFixableCases.None
+            if (!wrongNode.hasChildOfType(BLOCK) && !wrongNode.hasChildOfType(MODIFIER_LIST)) LongLineFixableCases.Fun(wrongNode) else LongLineFixableCases.None
 
     private fun checkComment(wrongNode: ASTNode, configuration: LineLengthConfiguration): LongLineFixableCases {
         val leftOffset = positionByOffset(wrongNode.startOffset).second
@@ -155,24 +199,16 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             }
         } else {
             val leftOffset = positionByOffset(newParent.findChildByType(STRING_TEMPLATE)!!.startOffset).second
-            if (newParent.findChildByType(STRING_TEMPLATE)!!.hasChildOfType(LONG_STRING_TEMPLATE_ENTRY)) {
-                val binList = wrongNode.findChildByType(STRING_TEMPLATE)!!.getChildren(null).filter { it.elementType in propertyList }
-                if (binList.size == 1) {
-                    return LongLineFixableCases.None
-                }
-                return LongLineFixableCases.PropertyWithTemplateEntry(wrongNode, configuration.lineLength, leftOffset, binList.toMutableList())
-            } else {
-                if (leftOffset > configuration.lineLength - STRING_PART_OFFSET) {
-                    return LongLineFixableCases.None
-                }
-                val text = wrongNode.findChildByType(STRING_TEMPLATE)!!.text.trim('"')
-                val lastCharIndex = configuration.lineLength.toInt() - leftOffset - STRING_PART_OFFSET
-                val indexLastSpace = (text.substring(0, lastCharIndex).lastIndexOf(' '))
-                if (indexLastSpace == -1) {
-                    return LongLineFixableCases.None
-                }
-                return LongLineFixableCases.Property(wrongNode, indexLastSpace, text)
+            if (leftOffset > configuration.lineLength - STRING_PART_OFFSET) {
+                return LongLineFixableCases.None
             }
+            val text = wrongNode.findChildByType(STRING_TEMPLATE)!!.text.trim('"')
+            val lastCharIndex = configuration.lineLength.toInt() - leftOffset - STRING_PART_OFFSET
+            val indexLastSpace = (text.substring(0, lastCharIndex).lastIndexOf(' '))
+            if (indexLastSpace == -1) {
+                return LongLineFixableCases.None
+            }
+            return LongLineFixableCases.Property(wrongNode, indexLastSpace, text)
         }
     }
 
@@ -189,57 +225,17 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     @Suppress("UnsafeCallOnNullableType", "WHEN_WITHOUT_ELSE")
-    private fun fixError(fixableType: LongLineFixableCases, configuration: LineLengthConfiguration) {
+    private fun fixError(fixableType: LongLineFixableCases) {
         when (fixableType) {
-            is LongLineFixableCases.Fun -> fixFunction(fixableType.node, configuration)
+            is LongLineFixableCases.Fun -> fixableType.node.appendNewlineMergingWhiteSpace(null, fixableType.node.findChildByType(EQ)!!.treeNext)
             is LongLineFixableCases.Comment -> fixComment(fixableType)
             is LongLineFixableCases.Condition -> fixLongBinaryExpression(fixableType)
             is LongLineFixableCases.Property -> createSplitProperty(fixableType)
-            is LongLineFixableCases.PropertyWithTemplateEntry -> fixLongString(fixableType)
+            is LongLineFixableCases.StringTemplate -> fixStringTemplate(fixableType)
+            is LongLineFixableCases.BinaryExpression -> fixBinaryExpression(fixableType.node)
             is LongLineFixableCases.None -> return
         }
     }
-
-    /**
-     * This function handles two cases:
-     *      1. When fun foo() = ...
-     *      2. There is an annotation before the function which breaks the maxLength parameter.
-     */
-    private fun fixFunction(wrongFunction: ASTNode, configuration: LineLengthConfiguration) {
-        if (wrongFunction.findChildByType(EQ) != null) {
-            wrongFunction.appendNewlineMergingWhiteSpace(null, wrongFunction.findChildByType(EQ)!!.treeNext)
-            return
-        } else if (wrongFunction.findChildByType(MODIFIER_LIST) != null) {
-            val annotationText = fixAnnotation(wrongFunction.findChildByType(MODIFIER_LIST)!!, configuration)
-            // The last one will always be fun declaration as there is no block.
-            val funText = wrongFunction.text.lines().last()
-            wrongFunction.treeParent.replaceChild(wrongFunction, KotlinParser().createNode("$annotationText\n$funText"))
-        }
-    }
-
-    private fun fixAnnotation(node: ASTNode, configuration: LineLengthConfiguration): String {
-        val annotationEntry = node.findChildByType(ANNOTATION_ENTRY)!!
-        val annotationText = annotationEntry.text
-        val firstPart = annotationText.substring(0, configuration.lineLength.toInt())
-        val remainingPart = annotationText.substring(configuration.lineLength.toInt(), annotationText.length)
-        return if (isDividerInString(firstPart)) {
-            divideString(firstPart) + remainingPart
-        } else {
-            divideAnnotationByComma(firstPart) + remainingPart
-        }
-    }
-
-    private fun divideAnnotationByComma(text: String): String {
-        val lastIndex = text.indexOfLast { it == ',' } + 1  // plus one is to include comma on the first line
-        return text.substring(0, lastIndex) + "\n" + text.substring(lastIndex, text.length)
-    }
-
-    private fun divideString(text: String): String {
-        val divisionIndex = text.indexOfLast { it == ' ' }
-        return text.substring(0, divisionIndex) + "\" +\n\"" + text.substring(divisionIndex, text.length)
-    }
-
-    private fun isDividerInString(text: String): Boolean = text.count { it == '"' } % 2 != 0
 
     private fun fixComment(wrongComment: LongLineFixableCases.Comment) {
         val wrongNode = wrongComment.node
@@ -251,6 +247,18 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
             addChild(LeafPsiElement(EOL_COMMENT, nodeText), wrongNode)
             removeChild(wrongNode)
         }
+    }
+
+    private fun fixBinaryExpression(node: ASTNode) {
+        val whiteSpaceAfterPlus = node.findChildByType(OPERATION_REFERENCE)!!.treeNext
+        node.replaceChild(whiteSpaceAfterPlus, PsiWhiteSpaceImpl("\n"))
+    }
+
+    private fun fixStringTemplate(wrongStringTemplate: LongLineFixableCases.StringTemplate) {
+        val incorrectText = wrongStringTemplate.node.text
+        val firstPart = incorrectText.substring(0, wrongStringTemplate.delimiterIndex)
+        val secondPart = incorrectText.substring(wrongStringTemplate.delimiterIndex, incorrectText.length)
+        wrongStringTemplate.node.treeParent.replaceChild(wrongStringTemplate.node, KotlinParser().createNode("$firstPart\" +\n\"$secondPart"))
     }
 
     /**
@@ -379,33 +387,6 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         }
     }
 
-    private fun fixLongString(wrongProperty: LongLineFixableCases.PropertyWithTemplateEntry) {
-        val wrongNode = wrongProperty.node
-        val leftOffset = wrongProperty.leftOffset
-        val lineLength = wrongProperty.maximumLineLength
-        val binList = wrongProperty.binList
-        val allText = binList.joinToString("") { it.text }
-        var binaryText = ""
-        binList.forEachIndexed { index, astNode ->
-            binaryText += astNode.text
-            if (STRING_PART_OFFSET + leftOffset + binaryText.length > lineLength) {
-                if (astNode.elementType == LITERAL_STRING_TEMPLATE_ENTRY) {
-                    val lastCharIndex = lineLength.toInt() - leftOffset - STRING_PART_OFFSET
-                    val indexLastSpace = (allText.substring(0, lastCharIndex).lastIndexOf(' '))
-                    if (indexLastSpace == -1) {
-                        return
-                    }
-                    splitTextAndCreateNode(wrongNode, allText, indexLastSpace)
-                    return
-                } else if (index == 0) {
-                    return
-                }
-                splitTextAndCreateNode(wrongNode, allText, binaryText.length)
-                return
-            }
-        }
-    }
-
     private fun createSplitProperty(wrongProperty: LongLineFixableCases.Property) {
         val node = wrongProperty.node
         val indexLastSpace = wrongProperty.indexLastSpace
@@ -442,6 +423,10 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
         object None : LongLineFixableCases()
 
         class Comment(val node: ASTNode, val indexLastSpace: Int) : LongLineFixableCases()
+
+        class StringTemplate(val node: ASTNode, val delimiterIndex: Int) : LongLineFixableCases()
+
+        class BinaryExpression(val node: ASTNode) : LongLineFixableCases()
 
         class Condition(
             val maximumLineLength: Long,

--- a/diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/LineLengthFixTest.kt
+++ b/diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/LineLengthFixTest.kt
@@ -48,6 +48,6 @@ class LineLengthFixTest : FixTestBase("test/paragraph3/long_line", ::LineLength)
 
     @Test
     fun `should fix annotation`() {
-        fixAndCompare("LongLineAnnotationExpected.kt", "LongLineAnnotationTest.kt", rulesConfigListShortLineLength)
+        fixAndCompare("LongLineAnnotationExpected.kt", "LongLineAnnotationTest.kt", rulesConfigListLineLength)
     }
 }

--- a/diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/LineLengthFixTest.kt
+++ b/diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/LineLengthFixTest.kt
@@ -45,4 +45,9 @@ class LineLengthFixTest : FixTestBase("test/paragraph3/long_line", ::LineLength)
     fun `shouldn't fix`() {
         fixAndCompare("LongExpressionNoFixExpected.kt", "LongExpressionNoFixTest.kt", rulesConfigListShortLineLength)
     }
+
+    @Test
+    fun `should fix annotation`() {
+        fixAndCompare("LongLineAnnotationExpected.kt", "LongLineAnnotationTest.kt", rulesConfigListShortLineLength)
+    }
 }

--- a/diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/LineLengthWarnTest.kt
+++ b/diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/LineLengthWarnTest.kt
@@ -19,6 +19,10 @@ class LineLengthWarnTest : LintTestBase(::LineLength) {
         RulesConfig(LONG_LINE.name, true,
             mapOf("lineLength" to "163"))
     )
+    private val shortLineLength: List<RulesConfig> = listOf(
+        RulesConfig(LONG_LINE.name, true,
+            mapOf("lineLength" to "40"))
+    )
     private val wrongUrl = "dhttps://www.google.com/search?q=djfhvkdfhvkdh+gthtdj%" +
             "3Bb&rlz=1C1GCEU_enRU909RU909&oq=posible+gthtdj%3Bb&aqs=chrome.." +
             "69i57j0l3.2680j1j7&sourceid=chrome&ie=UTF-8"
@@ -219,6 +223,20 @@ class LineLengthWarnTest : LintTestBase(::LineLength) {
                 """.trimMargin(),
             LintError(8, 1, ruleId, "${LONG_LINE.warnText()} max line length 120, but was 130", false),
             LintError(9, 1, ruleId, "${LONG_LINE.warnText()} max line length 120, but was 123", false)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.LONG_LINE)
+    fun `check annotation and fun with expr body`() {
+        lintMethod(
+            """
+                    |@Query(value = "ASDAASDASDASDASDASDASDASDAASDASDASDASDASDASDASDAASDASDASDASDASDASD")
+                    |fun foo() = println("ASDAASDASDASDASDASDASDASDAASDASDASDASDASDASDASDAASDASDASDASDASDASD")
+            """.trimMargin(),
+            LintError(1, 1, ruleId, "${LONG_LINE.warnText()} max line length 40, but was 84", false),
+            LintError(2, 1, ruleId, "${LONG_LINE.warnText()} max line length 40, but was 89", true),
+            rulesConfigList = shortLineLength
         )
     }
 }

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationExpected.kt
@@ -1,4 +1,17 @@
 package test.paragraph3.long_line
 
-@Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' and test_execution.test_suite_execution_id = ?3 limit ?1 offset ?2", nativeQuery = true)
-fun retrieveBatches(limit: Int, offset: Int, executionId: Long): List<Test>
+@Query(value = "select * from test inner join" +
+" test_execution on test.id = test_execution.test_id and test_execution.st", nativeQuery = true)
+fun retrieveBatches(limit: Int, offset: Int, executionId: Long): Some
+
+@Query(value = "select * from test inner join" +
+" test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' and test_execution.test_suite_execution_id = ?3 limit ?1 offset ?2", nativeQuery = true)
+fun some(limit: Int, offset: Int, executionId: Long): List<Test>
+
+@Query(value = "select * from test inner joi",
+ nativeQuery = true)
+fun test(limit: Int, offset: Int, executionId: Long): List<Test>
+
+@Query(value = "select * from test inner joibbb",
+ nativeQuery = true)
+fun cornerCase(limit: Int, offset: Int, executionId: Long): List<Test>

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationExpected.kt
@@ -1,0 +1,4 @@
+package test.paragraph3.long_line
+
+@Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' and test_execution.test_suite_execution_id = ?3 limit ?1 offset ?2", nativeQuery = true)
+fun retrieveBatches(limit: Int, offset: Int, executionId: Long): List<Test>

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationExpected.kt
@@ -8,10 +8,8 @@ fun retrieveBatches(limit: Int, offset: Int, executionId: Long): Some
 " test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' and test_execution.test_suite_execution_id = ?3 limit ?1 offset ?2", nativeQuery = true)
 fun some(limit: Int, offset: Int, executionId: Long): List<Test>
 
-@Query(value = "select * from test inner joi",
- nativeQuery = true)
+@Query(value = "select * from test inner joi", nativeQuery = true)
 fun test(limit: Int, offset: Int, executionId: Long): List<Test>
 
-@Query(value = "select * from test inner joibbb",
- nativeQuery = true)
+@Query(value = "select * from test inner joibbb", nativeQuery = true)
 fun cornerCase(limit: Int, offset: Int, executionId: Long): List<Test>

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationExpected.kt
@@ -13,3 +13,8 @@ fun test(limit: Int, offset: Int, executionId: Long): List<Test>
 
 @Query(value = "select * from test inner joibbb", nativeQuery = true)
 fun cornerCase(limit: Int, offset: Int, executionId: Long): List<Test>
+
+@Query(value = "select * from test inner join" +
+" test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' and test_execution.test_suite_execution_id = ?3 limit ?1 offset ?2", nativeQuery = true)
+fun some(limit: Int, offset: Int, executionId: Long) =
+ println("testtesttesttesttesttesttesttesttesttesttesttest")

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationTest.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationTest.kt
@@ -11,3 +11,6 @@ fun test(limit: Int, offset: Int, executionId: Long): List<Test>
 
 @Query(value = "select * from test inner joibbb", nativeQuery = true)
 fun cornerCase(limit: Int, offset: Int, executionId: Long): List<Test>
+
+@Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' and test_execution.test_suite_execution_id = ?3 limit ?1 offset ?2", nativeQuery = true)
+fun some(limit: Int, offset: Int, executionId: Long) = println("testtesttesttesttesttesttesttesttesttesttesttest")

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationTest.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationTest.kt
@@ -1,4 +1,13 @@
 package test.paragraph3.long_line
 
+@Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.st", nativeQuery = true)
+fun retrieveBatches(limit: Int, offset: Int, executionId: Long): Some
+
 @Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' and test_execution.test_suite_execution_id = ?3 limit ?1 offset ?2", nativeQuery = true)
-fun retrieveBatches(limit: Int, offset: Int, executionId: Long): List<Test>
+fun some(limit: Int, offset: Int, executionId: Long): List<Test>
+
+@Query(value = "select * from test inner joi", nativeQuery = true)
+fun test(limit: Int, offset: Int, executionId: Long): List<Test>
+
+@Query(value = "select * from test inner joibbb", nativeQuery = true)
+fun cornerCase(limit: Int, offset: Int, executionId: Long): List<Test>

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationTest.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineAnnotationTest.kt
@@ -1,0 +1,4 @@
+package test.paragraph3.long_line
+
+@Query(value = "select * from test inner join test_execution on test.id = test_execution.test_id and test_execution.status = 'READY' and test_execution.test_suite_execution_id = ?3 limit ?1 offset ?2", nativeQuery = true)
+fun retrieveBatches(limit: Int, offset: Int, executionId: Long): List<Test>

--- a/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineRValueExpected.kt
+++ b/diktat-rules/src/test/resources/test/paragraph3/long_line/LongLineRValueExpected.kt
@@ -22,10 +22,10 @@ fun foo() {
 " woooooordsdcsdcsdcsdc $variable"
 
     val longStringExpression = "First part" +
- "second Part"
+"second Part"
 
-    val longStringExpression = "First very long part" +
- "second Part"
+    val longStringExpression = "First very long" +
+" part" + "second Part"
 
     val veryLooooooooooooooooooooooooooooooongVal = "text"
 

--- a/diktat-rules/src/test/resources/test/smoke/src/main/kotlin/Example4Expected.kt
+++ b/diktat-rules/src/test/resources/test/smoke/src/main/kotlin/Example4Expected.kt
@@ -52,3 +52,9 @@ fun bar() {
     }
 }
 
+@Suppress("")
+fun foo() {
+    val y = "akgjsaujtmaksdkfasakgjsaujtmaksdkfasakgjsaujtmaksdkfasakgjsaujtm" +
+            " aksdkfasfasakgjsaujtmaksdfasafasakgjsaujtmaksdfasakgjsaujtmaksdfasakgjsaujtmaksdfasakgjsaujtmaksdfasakgjsaujtmaksdkgjsaujtmaksdfasakgjsaujtmaksd"
+}
+

--- a/diktat-rules/src/test/resources/test/smoke/src/main/kotlin/Example4Test.kt
+++ b/diktat-rules/src/test/resources/test/smoke/src/main/kotlin/Example4Test.kt
@@ -49,3 +49,8 @@ fun bar() {
     }
     diktatExtension.reporter = PlainReporter(System.out)
 }
+
+@Suppress("")
+fun foo() {
+    val y = "akgjsaujtmaksdkfasakgjsaujtmaksdkfasakgjsaujtmaksdkfasakgjsaujtm aksdkfasfasakgjsaujtmaksdfasafasakgjsaujtmaksdfasakgjsaujtmaksdfasakgjsaujtmaksdfasakgjsaujtmaksdfasakgjsaujtmaksdkgjsaujtmaksdfasakgjsaujtmaksd"
+}

